src/main/kotlin/no/nav/syfo/cronjob/virksomhetsnavn/PersonOppfolgingstilfelleVirksomhetsnavnService.kt

```diff
@@ -0,0 +1,29 @@
+package no.nav.syfo.cronjob.virksomhetsnavn
+
+import no.nav.syfo.application.database.DatabaseInterface
+import no.nav.syfo.client.ereg.EregClient
+import no.nav.syfo.domain.Virksomhetsnummer
+import java.util.*
+
+class PersonOppfolgingstilfelleVirksomhetsnavnService(
+    private val database: DatabaseInterface,
+    private val eregClient: EregClient,
+) {
+    fun getPersonOppfolgingstilfelleVirksomhetMissingVirksomhetsnavnList(): List<Pair<Int, Virksomhetsnummer>> =
+        database.getPersonOppfolgingstilfelleVirksomhetMissingVirksomhetsnavnList()
+
+    suspend fun updateVirksomhetsnavn(
+        idVirksomhetsnummerPair: Pair<Int, Virksomhetsnummer>,
+    ) {
+        val (id, virksomhetsnummer) = idVirksomhetsnummerPair
+        eregClient.organisasjonVirksomhetsnavn(
+            callId = UUID.randomUUID().toString(),
+            virksomhetsnummer = virksomhetsnummer,
+        )?.let { eregOrganisasjonVirksomhetsnavn ->
+            database.updatePersonOppfolgingstilfelleVirksomhetVirksomhetsnavn(
+                personOppfolgingstilfelleVirksomhetId = id,
+                virksomhetsnavn = eregOrganisasjonVirksomhetsnavn.virksomhetsnavn,
+            )
+        } ?: throw RuntimeException("Failed to store Virksomhetsnavn: response from Ereg missing")
```

Comment:
Trenger man egentlig å kaste exception her? Det er jo ikke krise om vi mangler navn? 🤔

Reply:
Hvis ikke så catches det ikke noe i funksjonen over og vi får ikke oversikt over hvor mange som feiler.
Tenker på telleren result.failed i kallende funksjon hvor det er en try/catch.

Reply:
Det kunne jo vært løst med å returnere/sjekke `null`? Ikke at det er krise med exception, men det er heller ikke nødvendig hvis dette ikke er en eksepsjonell tilstand?

Reply:
Returnere Boolean false i stedet for exception mener du?